platform/tests/TESTS/mbed_functional/callback/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.19.0 FATAL_ERROR)
+
+set(MBED_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../../../.. CACHE INTERNAL "")
+set(MBED_CONFIG_PATH ${CMAKE_CURRENT_SOURCE_DIR}/.mbedbuild CACHE INTERNAL "")
+set(APP_TARGET mbed-platform-mbed-functional-callback)
+
+include(${MBED_PATH}/tools/cmake/app.cmake)
+
+add_subdirectory(${MBED_PATH} build)
+
+add_executable(${APP_TARGET})
+
+mbed_configure_app_target(${APP_TARGET})
+
+mbed_set_mbed_target_linker_script(${APP_TARGET})
+
+project(${APP_TARGET})
+
+target_sources(${APP_TARGET}
+    PRIVATE
+        main.cpp
+)
+
+if(MBED_BAREMETAL_GREENTEA_TEST)
+	set(USE_LIB mbed-baremetal)
```

Comment:
```suggestion
	list(APPEND mbed_libs mbed-baremetal)
```

Reply:
@hugueskamba As we are linking only two libraries, don't see any advantage for creating a list, and directly specifying required libraries in target_link_libraries for APP_TARGET gives more visibility and readability. In all our examples application CMake, we have specified the required libraries directly in the target_link_libraries instead of putting in the list, test suite CMake is also similar kind of example CMake

Reply:
The example projects you are referring to do not have to make a choice of library. Here you are selecting `mbed-os` or `mbed-baremetal` and creating a variable to store the library to use for the project then you add an extra library.

IMO it makes more sense to use the variable (that you have created anyway) to store all libs and then just add that in `target_link_libraries()`.

Also can you not just use a macro to pass what's changing between the applications?